pkg/pillar/cmd/zedagent/handlecertconfig.go

```diff
@@ -335,6 +329,35 @@ func getCertsFromController(ctx *zedagentContext, desc string) (success bool) {
 	return true
 }
 
+// Fetch and verify the controller certificates. Returns true if certificates
+// have not changed or the update was successfully applied. False is returned
+// if the function failed to fetch/verify/unmarshal certs.
+//
+// If main controller is unavailable (@false is returned), the next set of
+// attempts is to fallback to retrieve certs from the LOC.
```

Comment:
For the EdgeDevConfig we try to fetch from both locations and compare the timestamp to make sure we don't accidentally accept an older config. Do we need the same type of checks for the certs?

Reply:
Yes, this makes a lot of sense to add this check.

Reply:
Alternatively save the certs we receive from both but that would require retry logic since today EVE assumes there is one controller signing certificate and one controller/ECDH certificate for object encryption.

Given that, unlike the EdgeDevConfig, we don't have a timestamp in the payload returned by the certificate API, we can't add a simple compare. So this probably requires some more thought. Could wrap up this PR first if you prefer.

Reply:
@eriknordmark PEM certificates we are using have the following fields:
```
NotBefore, NotAfter [time](https://pkg.go.dev/time).[Time](https://pkg.go.dev/time#Time) // Validity bounds.
```
(https://pkg.go.dev/crypto/x509#Certificate)

Assuming we always moving forward with certificates, we apply a new certificate only if it is not older:
```
if newCert.NotBefore <= oldCert.NotBefore:
       reject
```

This still gives us freedom to correct mistakes if new certificates were issues incorrectly by keeping the `NotAfter` date same and moving the `NotBefore` forward. 

This change is relatively small and can be done in a separate PR.

Reply:
Very good suggestion.